2021-11-08 11:10:00,580 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.shouldTryMultipleAddresses() throws java.lang.Exception]
2021-11-08 11:10:00,586 [myid:] - INFO  [main:Learner@127] - leaderConnectDelayDuringRetryMs: 100
2021-11-08 11:10:00,587 [myid:] - INFO  [main:Learner@128] - TCP NoDelay set to: true
2021-11-08 11:10:00,587 [myid:] - INFO  [main:Learner@129] - learner.asyncSending = false
2021-11-08 11:10:00,587 [myid:] - INFO  [main:Learner@130] - learner.closeSocketAsync = false
2021-11-08 11:10:00,593 [myid:] - INFO  [main:QuorumPeer@796] - zookeeper.quorumCnxnTimeoutMs=-1
2021-11-08 11:10:00,600 [myid:] - INFO  [main:X509Util@77] - Setting -D jdk.tls.rejectClientInitiatedRenegotiation=true to disable client-initiated TLS renegotiation
2021-11-08 11:10:00,602 [myid:] - INFO  [main:QuorumPeer@1720] - tickTime set to 2000
2021-11-08 11:10:00,602 [myid:] - INFO  [main:QuorumPeer@1800] - initLimit set to 5
2021-11-08 11:10:00,603 [myid:] - INFO  [main:QuorumPeer@1987] - syncLimit set to 2
2021-11-08 11:10:01,140 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,140 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,140 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,140 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,255 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,256 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,256 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,255 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,363 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@445] - Unexpected exception, tries=2, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,363 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@445] - Unexpected exception, tries=2, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,363 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=2, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,363 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@445] - Unexpected exception, tries=2, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,473 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=3, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,473 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@445] - Unexpected exception, tries=3, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,473 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@445] - Unexpected exception, tries=3, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,473 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@445] - Unexpected exception, tries=3, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@437] - Unexpected exception, retries exceeded. tries=4, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@437] - Unexpected exception, retries exceeded. tries=4, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@437] - Unexpected exception, retries exceeded. tries=4, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@437] - Unexpected exception, retries exceeded. tries=4, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,582 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,583 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:01,591 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.shouldTryMultipleAddresses() throws java.lang.Exception]
2021-11-08 11:10:01,596 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.multipleAddressesSomeAreFailing() throws java.lang.Exception]
2021-11-08 11:10:01,596 [myid:] - INFO  [main:QuorumPeer@1720] - tickTime set to 2000
2021-11-08 11:10:01,596 [myid:] - INFO  [main:QuorumPeer@1800] - initLimit set to 5
2021-11-08 11:10:01,596 [myid:] - INFO  [main:QuorumPeer@1987] - syncLimit set to 2
2021-11-08 11:10:02,005 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:3333
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:02,005 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:4444
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:02,005 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:2222
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:02,008 [myid:] - INFO  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@381] - Successfully connected to leader, using address: 0.0.0.0/0.0.0.0:1111
2021-11-08 11:10:02,114 [myid:] - INFO  [LeaderConnector-0.0.0.0/0.0.0.0:3333:Learner$LeaderConnector@383] - Connection to the leader is already established, close the redundant connection
2021-11-08 11:10:02,114 [myid:] - INFO  [LeaderConnector-0.0.0.0/0.0.0.0:2222:Learner$LeaderConnector@383] - Connection to the leader is already established, close the redundant connection
2021-11-08 11:10:02,114 [myid:] - INFO  [LeaderConnector-0.0.0.0/0.0.0.0:4444:Learner$LeaderConnector@383] - Connection to the leader is already established, close the redundant connection
2021-11-08 11:10:02,117 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.multipleAddressesSomeAreFailing() throws java.lang.Exception]
2021-11-08 11:10:02,118 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.truncFailTest() throws java.lang.Exception]
2021-11-08 11:10:02,517 [myid:] - INFO  [main:FileTxnSnapLog@124] - zookeeper.snapshot.trust.empty : false
2021-11-08 11:10:02,527 [myid:] - INFO  [main:ZookeeperBanner@42] - 
2021-11-08 11:10:02,527 [myid:] - INFO  [main:ZookeeperBanner@42] -   ______                  _                                          
2021-11-08 11:10:02,527 [myid:] - INFO  [main:ZookeeperBanner@42] -  |___  /                 | |                                         
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] -     / /    ___     ___   | | __   ___    ___   _ __     ___   _ __   
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] -    / /    / _ \   / _ \  | |/ /  / _ \  / _ \ | '_ \   / _ \ | '__|
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] -   / /__  | (_) | | (_) | |   <  |  __/ |  __/ | |_) | |  __/ | |    
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] -  /_____|  \___/   \___/  |_|\_\  \___|  \___| | .__/   \___| |_|
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] -                                               | |                     
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] -                                               |_|                     
2021-11-08 11:10:02,528 [myid:] - INFO  [main:ZookeeperBanner@42] - 
2021-11-08 11:10:11,580 [myid:] - INFO  [main:Environment@98] - Server environment:zookeeper.version=3.7.0-02d8e7728613600d355308bb0a256b6f5b41360c-dirty, built on 2021-11-08 03:06 UTC
2021-11-08 11:10:11,580 [myid:] - INFO  [main:Environment@98] - Server environment:host.name=DESKTOP-3V3H020
2021-11-08 11:10:11,581 [myid:] - INFO  [main:Environment@98] - Server environment:java.version=1.8.0_31
2021-11-08 11:10:11,581 [myid:] - INFO  [main:Environment@98] - Server environment:java.vendor=Oracle Corporation
2021-11-08 11:10:11,581 [myid:] - INFO  [main:Environment@98] - Server environment:java.home=C:\Program Files\Java\jdk1.8.0_31\jre
2021-11-08 11:10:11,581 [myid:] - INFO  [main:Environment@98] - Server environment:java.class.path=D:\mysoft\faiscoGit\myStudy\ZooKeeperLearning\apache-zookeeper-3.7.0\zookeeper-server\target\test-classes;D:\mysoft\faiscoGit\myStudy\ZooKeeperLearning\apache-zookeeper-3.7.0\zookeeper-server\target\classes;D:\mysoft\maven\repository\com\github\spotbugs\spotbugs-annotations\4.0.2\spotbugs-annotations-4.0.2.jar;D:\mysoft\maven\repository\com\google\code\findbugs\jsr305\3.0.2\jsr305-3.0.2.jar;D:\mysoft\maven\repository\org\hamcrest\hamcrest-library\2.2\hamcrest-library-2.2.jar;D:\mysoft\maven\repository\org\hamcrest\hamcrest-core\2.2\hamcrest-core-2.2.jar;D:\mysoft\maven\repository\org\hamcrest\hamcrest\2.2\hamcrest-2.2.jar;D:\mysoft\maven\repository\org\apache\commons\commons-collections4\4.4\commons-collections4-4.4.jar;D:\mysoft\maven\repository\org\apache\zookeeper\zookeeper-jute\3.7.0\zookeeper-jute-3.7.0.jar;D:\mysoft\maven\repository\commons-cli\commons-cli\1.4\commons-cli-1.4.jar;D:\mysoft\maven\repository\org\apache\yetus\audience-annotations\0.12.0\audience-annotations-0.12.0.jar;D:\mysoft\maven\repository\io\netty\netty-handler\4.1.59.Final\netty-handler-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-common\4.1.59.Final\netty-common-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-resolver\4.1.59.Final\netty-resolver-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-buffer\4.1.59.Final\netty-buffer-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-transport\4.1.59.Final\netty-transport-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-codec\4.1.59.Final\netty-codec-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-transport-native-epoll\4.1.59.Final\netty-transport-native-epoll-4.1.59.Final.jar;D:\mysoft\maven\repository\io\netty\netty-transport-native-unix-common\4.1.59.Final\netty-transport-native-unix-common-4.1.59.Final.jar;D:\mysoft\maven\repository\org\slf4j\slf4j-api\1.7.30\slf4j-api-1.7.30.jar;D:\mysoft\maven\repository\org\slf4j\slf4j-log4j12\1.7.30\slf4j-log4j12-1.7.30.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-server\9.4.38.v20210224\jetty-server-9.4.38.v20210224.jar;D:\mysoft\maven\repository\javax\servlet\javax.servlet-api\3.1.0\javax.servlet-api-3.1.0.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-http\9.4.38.v20210224\jetty-http-9.4.38.v20210224.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-util\9.4.38.v20210224\jetty-util-9.4.38.v20210224.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-io\9.4.38.v20210224\jetty-io-9.4.38.v20210224.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-servlet\9.4.38.v20210224\jetty-servlet-9.4.38.v20210224.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-security\9.4.38.v20210224\jetty-security-9.4.38.v20210224.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-util-ajax\9.4.38.v20210224\jetty-util-ajax-9.4.38.v20210224.jar;D:\mysoft\maven\repository\org\eclipse\jetty\jetty-client\9.4.38.v20210224\jetty-client-9.4.38.v20210224.jar;D:\mysoft\maven\repository\com\fasterxml\jackson\core\jackson-databind\2.10.5.1\jackson-databind-2.10.5.1.jar;D:\mysoft\maven\repository\com\fasterxml\jackson\core\jackson-annotations\2.10.5\jackson-annotations-2.10.5.jar;D:\mysoft\maven\repository\com\fasterxml\jackson\core\jackson-core\2.10.5\jackson-core-2.10.5.jar;D:\mysoft\maven\repository\org\bouncycastle\bcprov-jdk15on\1.60\bcprov-jdk15on-1.60.jar;D:\mysoft\maven\repository\org\bouncycastle\bcpkix-jdk15on\1.60\bcpkix-jdk15on-1.60.jar;D:\mysoft\maven\repository\jline\jline\2.14.6\jline-2.14.6.jar;D:\mysoft\maven\repository\io\dropwizard\metrics\metrics-core\4.1.12.1\metrics-core-4.1.12.1.jar;D:\mysoft\maven\repository\log4j\log4j\1.2.17\log4j-1.2.17.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-core\2.0.0\kerb-core-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerby-pkix\2.0.0\kerby-pkix-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerby-asn1\2.0.0\kerby-asn1-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerby-util\2.0.0\kerby-util-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-simplekdc\2.0.0\kerb-simplekdc-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-client\2.0.0\kerb-client-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-common\2.0.0\kerb-common-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-crypto\2.0.0\kerb-crypto-2.0.0.jar;D:\mysoft\maven\repository\commons-io\commons-io\2.6\commons-io-2.6.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-util\2.0.0\kerb-util-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\token-provider\2.0.0\token-provider-2.0.0.jar;D:\mysoft\maven\repository\com\nimbusds\nimbus-jose-jwt\4.41.2\nimbus-jose-jwt-4.41.2.jar;D:\mysoft\maven\repository\com\github\stephenc\jcip\jcip-annotations\1.0-1\jcip-annotations-1.0-1.jar;D:\mysoft\maven\repository\net\minidev\json-smart\2.3\json-smart-2.3.jar;D:\mysoft\maven\repository\net\minidev\accessors-smart\1.2\accessors-smart-1.2.jar;D:\mysoft\maven\repository\org\ow2\asm\asm\5.0.4\asm-5.0.4.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-admin\2.0.0\kerb-admin-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-server\2.0.0\kerb-server-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerb-identity\2.0.0\kerb-identity-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerby-xdr\2.0.0\kerby-xdr-2.0.0.jar;D:\mysoft\maven\repository\org\apache\kerby\kerby-config\2.0.0\kerby-config-2.0.0.jar;D:\mysoft\maven\repository\org\mockito\mockito-core\3.6.28\mockito-core-3.6.28.jar;D:\mysoft\maven\repository\net\bytebuddy\byte-buddy\1.10.18\byte-buddy-1.10.18.jar;D:\mysoft\maven\repository\net\bytebuddy\byte-buddy-agent\1.10.18\byte-buddy-agent-1.10.18.jar;D:\mysoft\maven\repository\org\objenesis\objenesis\3.1\objenesis-3.1.jar;D:\mysoft\maven\repository\org\jmockit\jmockit\1.48\jmockit-1.48.jar;D:\mysoft\maven\repository\org\junit\jupiter\junit-jupiter-api\5.6.2\junit-jupiter-api-5.6.2.jar;D:\mysoft\maven\repository\org\apiguardian\apiguardian-api\1.1.0\apiguardian-api-1.1.0.jar;D:\mysoft\maven\repository\org\opentest4j\opentest4j\1.2.0\opentest4j-1.2.0.jar;D:\mysoft\maven\repository\org\junit\platform\junit-platform-commons\1.6.2\junit-platform-commons-1.6.2.jar;D:\mysoft\maven\repository\org\junit\jupiter\junit-jupiter-engine\5.6.2\junit-jupiter-engine-5.6.2.jar;D:\mysoft\maven\repository\org\junit\platform\junit-platform-engine\1.6.2\junit-platform-engine-1.6.2.jar;D:\mysoft\maven\repository\org\junit\jupiter\junit-jupiter-params\5.6.2\junit-jupiter-params-5.6.2.jar;D:\mysoft\maven\repository\org\junit\platform\junit-platform-runner\1.6.2\junit-platform-runner-1.6.2.jar;D:\mysoft\maven\repository\junit\junit\4.13\junit-4.13.jar;D:\mysoft\maven\repository\org\junit\platform\junit-platform-launcher\1.6.2\junit-platform-launcher-1.6.2.jar;D:\mysoft\maven\repository\org\junit\platform\junit-platform-suite-api\1.6.2\junit-platform-suite-api-1.6.2.jar;D:\mysoft\maven\repository\org\xerial\snappy\snappy-java\1.1.7.7\snappy-java-1.1.7.7.jar;
2021-11-08 11:10:11,581 [myid:] - INFO  [main:Environment@98] - Server environment:java.library.path=C:\Program Files\Java\jdk1.8.0_31\jre\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\Java\jdk1.8.0_31\jre\bin;C:\Python39\Scripts\;C:\Python39\;C:\Python27\;C:\Python27\Scripts;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\iCLS\;C:\Program Files\Intel\Intel(R) Management Engine Components\iCLS\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\TortoiseSVN\bin;C:\Program Files\Java\jdk1.8.0_31\bin;D:\soft\Git\cmd;D:\mysoft\maven\apache-maven-3.6.1\bin;C:\ProgramData\chocolatey\bin;C:\Program Files\nodejs\;D:\mysoft\gradle-7.0\bin;C:\ProgramData\Oracle\Java\javapath;C:\Program Files\OpenVPN\bin;C:\Users\faisco\AppData\Local\Microsoft\WindowsApps;;D:\soft\Microsoft VS Code\bin;D:\soft\PyCharm Community Edition 2019.1.3\bin;;D:\soft\PyCharm 2019.2.3\bin;;C:\Users\faisco\AppData\Local\Microsoft\WindowsApps;C:\Program Files\JetBrains\IntelliJ IDEA 2020.1\bin;;C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.1.3\bin;;C:\Users\faisco\AppData\Roaming\npm;.
2021-11-08 11:10:11,581 [myid:] - INFO  [main:Environment@98] - Server environment:java.io.tmpdir=C:\Users\faisco\AppData\Local\Temp\
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:java.compiler=<NA>
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:os.name=Windows 8.1
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:os.arch=amd64
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:os.version=6.3
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:user.name=faisco
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:user.home=C:\Users\faisco
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:user.dir=D:\mysoft\faiscoGit\myStudy\ZooKeeperLearning\apache-zookeeper-3.7.0\zookeeper-server
2021-11-08 11:10:11,582 [myid:] - INFO  [main:Environment@98] - Server environment:os.memory.free=213MB
2021-11-08 11:10:11,583 [myid:] - INFO  [main:Environment@98] - Server environment:os.memory.max=455MB
2021-11-08 11:10:11,583 [myid:] - INFO  [main:Environment@98] - Server environment:os.memory.total=243MB
2021-11-08 11:10:11,583 [myid:] - INFO  [main:ZooKeeperServer@138] - zookeeper.enableEagerACLCheck = false
2021-11-08 11:10:11,583 [myid:] - INFO  [main:ZooKeeperServer@151] - zookeeper.digest.enabled = true
2021-11-08 11:10:11,583 [myid:] - INFO  [main:ZooKeeperServer@155] - zookeeper.closeSessionTxn.enabled = true
2021-11-08 11:10:11,584 [myid:] - INFO  [main:ZooKeeperServer@1499] - zookeeper.flushDelay=0
2021-11-08 11:10:11,584 [myid:] - INFO  [main:ZooKeeperServer@1508] - zookeeper.maxWriteQueuePollTime=0
2021-11-08 11:10:11,584 [myid:] - INFO  [main:ZooKeeperServer@1517] - zookeeper.maxBatchSize=1000
2021-11-08 11:10:11,584 [myid:] - INFO  [main:ZooKeeperServer@260] - zookeeper.intBufferStartingSizeBytes = 1024
2021-11-08 11:10:11,597 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:11,597 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:11,597 [myid:] - INFO  [main:ZKDatabase@133] - zookeeper.snapshotSizeFactor = 0.33
2021-11-08 11:10:11,598 [myid:] - INFO  [main:ZKDatabase@153] - zookeeper.commitLogCount=500
2021-11-08 11:10:11,599 [myid:] - INFO  [main:BlueThrottle@141] - Weighed connection throttling is disabled
2021-11-08 11:10:11,600 [myid:] - INFO  [main:ZooKeeperServer@1300] - minSessionTimeout set to 2000
2021-11-08 11:10:11,601 [myid:] - INFO  [main:ZooKeeperServer@1309] - maxSessionTimeout set to 2000
2021-11-08 11:10:11,602 [myid:] - INFO  [main:ResponseCache@45] - getData response cache size is initialized with value 400.
2021-11-08 11:10:11,602 [myid:] - INFO  [main:ResponseCache@45] - getChildren response cache size is initialized with value 400.
2021-11-08 11:10:11,603 [myid:] - INFO  [main:RequestPathMetricsCollector@109] - zookeeper.pathStats.slotCapacity = 60
2021-11-08 11:10:11,603 [myid:] - INFO  [main:RequestPathMetricsCollector@110] - zookeeper.pathStats.slotDuration = 15
2021-11-08 11:10:11,603 [myid:] - INFO  [main:RequestPathMetricsCollector@111] - zookeeper.pathStats.maxDepth = 6
2021-11-08 11:10:11,603 [myid:] - INFO  [main:RequestPathMetricsCollector@112] - zookeeper.pathStats.initialDelay = 5
2021-11-08 11:10:11,603 [myid:] - INFO  [main:RequestPathMetricsCollector@113] - zookeeper.pathStats.delay = 5
2021-11-08 11:10:11,603 [myid:] - INFO  [main:RequestPathMetricsCollector@114] - zookeeper.pathStats.enabled = false
2021-11-08 11:10:11,604 [myid:] - INFO  [main:ZooKeeperServer@1536] - The max bytes for all large requests are set to 104857600
2021-11-08 11:10:11,605 [myid:] - INFO  [main:ZooKeeperServer@1550] - The large request threshold is set to -1
2021-11-08 11:10:11,605 [myid:] - INFO  [main:AuthenticationHelper@66] - zookeeper.enforce.auth.enabled = false
2021-11-08 11:10:11,605 [myid:] - INFO  [main:AuthenticationHelper@67] - zookeeper.enforce.auth.schemes = []
2021-11-08 11:10:11,605 [myid:] - INFO  [main:ZooKeeperServer@361] - Created server with tickTime 2000 minSessionTimeout 2000 maxSessionTimeout 2000 clientPortListenBacklog -1 datadir src\test\resources\data\test2327331111373633408.dir\version-2 snapdir src\test\resources\data\test2327331111373633408.dir\version-2
2021-11-08 11:10:11,606 [myid:] - INFO  [main:QuorumPeer@920] - Peer state changed: looking - trunc
2021-11-08 11:10:11,607 [myid:] - WARN  [main:Learner@596] - Truncating log to get in sync with the leader 0x0
2021-11-08 11:10:11,607 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:11,607 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:11,609 [myid:] - ERROR [main:FileTxnSnapLog@530] - Unable to truncate Txn log
java.io.IOException: No log files found to truncate! This could happen if you still have snapshots from an old setup or log files were deleted accidentally or dataLogDir was changed in zoo.cfg.
	at org.apache.zookeeper.server.persistence.FileTxnLog.truncate(FileTxnLog.java:459)
	at org.apache.zookeeper.server.persistence.FileTxnSnapLog.truncateLog(FileTxnSnapLog.java:518)
	at org.apache.zookeeper.server.ZKDatabase.truncateLog(ZKDatabase.java:603)
	at org.apache.zookeeper.server.quorum.Learner.syncWithLeader(Learner.java:597)
	at org.apache.zookeeper.server.quorum.LearnerTest.truncFailTest(LearnerTest.java:356)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:686)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$106/1045941616.apply(Unknown Source)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall$$Lambda$107/1800890735.apply(Unknown Source)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$$Lambda$231/1587819720.apply(Unknown Source)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:212)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$266/919112242.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:208)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$177/2142080121.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$176/673186785.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$175/849373393.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$181/300031246.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$177/2142080121.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$176/673186785.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$175/849373393.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService$$Lambda$181/300031246.accept(Unknown Source)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$177/2142080121.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$176/673186785.invoke(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask$$Lambda$175/849373393.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$141/212921632.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:113)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
2021-11-08 11:10:11,610 [myid:] - ERROR [main:Learner@600] - Not able to truncate the log 0x0
2021-11-08 11:10:11,616 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.truncFailTest() throws java.lang.Exception]
2021-11-08 11:10:11,617 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.connectionRetryTimeoutTest() throws java.lang.Exception]
2021-11-08 11:10:11,618 [myid:] - INFO  [main:QuorumPeer@1720] - tickTime set to 2000
2021-11-08 11:10:11,618 [myid:] - INFO  [main:QuorumPeer@1800] - initLimit set to 5
2021-11-08 11:10:11,618 [myid:] - INFO  [main:QuorumPeer@1987] - syncLimit set to 2
2021-11-08 11:10:11,619 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:11,721 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:11,831 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=2, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:11,941 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=3, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,053 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@437] - Unexpected exception, retries exceeded. tries=4, remaining init limit=10000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,053 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,054 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.connectionRetryTimeoutTest() throws java.lang.Exception]
2021-11-08 11:10:12,056 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.connectionInitLimitTimeoutTest() throws java.lang.Exception]
2021-11-08 11:10:12,056 [myid:] - INFO  [main:QuorumPeer@1720] - tickTime set to 2000
2021-11-08 11:10:12,056 [myid:] - INFO  [main:QuorumPeer@1800] - initLimit set to 5
2021-11-08 11:10:12,056 [myid:] - INFO  [main:QuorumPeer@1987] - syncLimit set to 2
2021-11-08 11:10:12,057 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=6000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,161 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=2000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,275 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@429] - Unexpected exception, connectToLeader exceeded. tries=2, remaining init limit=-2000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,275 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,276 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.connectionInitLimitTimeoutTest() throws java.lang.Exception]
2021-11-08 11:10:12,277 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.connectToLearnerMasterLimitTest() throws java.lang.Exception]
2021-11-08 11:10:12,277 [myid:] - INFO  [main:QuorumPeer@1720] - tickTime set to 2000
2021-11-08 11:10:12,277 [myid:] - INFO  [main:QuorumPeer@1800] - initLimit set to 2
2021-11-08 11:10:12,277 [myid:] - INFO  [main:QuorumPeer@1987] - syncLimit set to 2
2021-11-08 11:10:12,277 [myid:] - INFO  [main:QuorumPeer@2002] - connectToLearnerMasterLimit set to 5
2021-11-08 11:10:12,278 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=0, remaining init limit=6000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,381 [myid:] - WARN  [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@445] - Unexpected exception, tries=1, remaining init limit=2000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,491 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@429] - Unexpected exception, connectToLeader exceeded. tries=2, remaining init limit=-2000, connecting to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,491 [myid:] - ERROR [LeaderConnector-0.0.0.0/0.0.0.0:1111:Learner$LeaderConnector@389] - Failed connect to 0.0.0.0/0.0.0.0:1111
java.io.IOException: Test injected Socket.connect() error.
	at org.apache.zookeeper.server.quorum.LearnerTest$TestLearner.sockConnect(LearnerTest.java:125)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.connectToLeader(Learner.java:419)
	at org.apache.zookeeper.server.quorum.Learner$LeaderConnector.run(Learner.java:377)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2021-11-08 11:10:12,492 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.connectToLearnerMasterLimitTest() throws java.lang.Exception]
2021-11-08 11:10:12,493 [myid:] - INFO  [main:ZKTestCase@77] - STARTING Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.syncTest() throws java.lang.Exception]
2021-11-08 11:10:12,498 [myid:] - INFO  [main:FileTxnSnapLog@124] - zookeeper.snapshot.trust.empty : false
2021-11-08 11:10:12,500 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:12,501 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:12,501 [myid:] - INFO  [main:ZKDatabase@133] - zookeeper.snapshotSizeFactor = 0.33
2021-11-08 11:10:12,501 [myid:] - INFO  [main:ZKDatabase@153] - zookeeper.commitLogCount=500
2021-11-08 11:10:12,501 [myid:] - INFO  [main:ZooKeeperServer@1300] - minSessionTimeout set to 2000
2021-11-08 11:10:12,502 [myid:] - INFO  [main:ZooKeeperServer@1309] - maxSessionTimeout set to 2000
2021-11-08 11:10:12,502 [myid:] - INFO  [main:ResponseCache@45] - getData response cache size is initialized with value 400.
2021-11-08 11:10:12,502 [myid:] - INFO  [main:ResponseCache@45] - getChildren response cache size is initialized with value 400.
2021-11-08 11:10:12,502 [myid:] - INFO  [main:RequestPathMetricsCollector@109] - zookeeper.pathStats.slotCapacity = 60
2021-11-08 11:10:12,503 [myid:] - INFO  [main:RequestPathMetricsCollector@110] - zookeeper.pathStats.slotDuration = 15
2021-11-08 11:10:12,503 [myid:] - INFO  [main:RequestPathMetricsCollector@111] - zookeeper.pathStats.maxDepth = 6
2021-11-08 11:10:12,503 [myid:] - INFO  [main:RequestPathMetricsCollector@112] - zookeeper.pathStats.initialDelay = 5
2021-11-08 11:10:12,503 [myid:] - INFO  [main:RequestPathMetricsCollector@113] - zookeeper.pathStats.delay = 5
2021-11-08 11:10:12,503 [myid:] - INFO  [main:RequestPathMetricsCollector@114] - zookeeper.pathStats.enabled = false
2021-11-08 11:10:12,503 [myid:] - INFO  [main:ZooKeeperServer@1536] - The max bytes for all large requests are set to 104857600
2021-11-08 11:10:12,504 [myid:] - INFO  [main:ZooKeeperServer@1550] - The large request threshold is set to -1
2021-11-08 11:10:12,504 [myid:] - INFO  [main:AuthenticationHelper@66] - zookeeper.enforce.auth.enabled = false
2021-11-08 11:10:12,504 [myid:] - INFO  [main:AuthenticationHelper@67] - zookeeper.enforce.auth.schemes = []
2021-11-08 11:10:12,504 [myid:] - INFO  [main:ZooKeeperServer@361] - Created server with tickTime 2000 minSessionTimeout 2000 maxSessionTimeout 2000 clientPortListenBacklog -1 datadir src\test\resources\data\test2414824585606454936.dir\version-2 snapdir src\test\resources\data\test2414824585606454936.dir\version-2
2021-11-08 11:10:12,510 [myid:] - INFO  [main:QuorumPeer@920] - Peer state changed: looking - snap
2021-11-08 11:10:12,510 [myid:] - INFO  [main:Learner@573] - Getting a snapshot from leader 0x0
2021-11-08 11:10:12,510 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:12,510 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:12,514 [myid:] - WARN  [main:Learner@634] - Got zxid 0x0 expected 0x1
2021-11-08 11:10:12,516 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:12,516 [myid:] - INFO  [main:WatchManagerFactory@42] - Using org.apache.zookeeper.server.watch.WatchManager as watch manager
2021-11-08 11:10:12,516 [myid:] - INFO  [main:ZKDatabase@133] - zookeeper.snapshotSizeFactor = 0.33
2021-11-08 11:10:12,516 [myid:] - INFO  [main:ZKDatabase@153] - zookeeper.commitLogCount=500
2021-11-08 11:10:12,516 [myid:] - INFO  [main:ZooKeeperServer@1300] - minSessionTimeout set to 2000
2021-11-08 11:10:12,516 [myid:] - INFO  [main:ZooKeeperServer@1309] - maxSessionTimeout set to 2000
2021-11-08 11:10:12,517 [myid:] - INFO  [main:ResponseCache@45] - getData response cache size is initialized with value 400.
2021-11-08 11:10:12,517 [myid:] - INFO  [main:ResponseCache@45] - getChildren response cache size is initialized with value 400.
2021-11-08 11:10:12,517 [myid:] - INFO  [main:RequestPathMetricsCollector@109] - zookeeper.pathStats.slotCapacity = 60
2021-11-08 11:10:12,517 [myid:] - INFO  [main:RequestPathMetricsCollector@110] - zookeeper.pathStats.slotDuration = 15
2021-11-08 11:10:12,517 [myid:] - INFO  [main:RequestPathMetricsCollector@111] - zookeeper.pathStats.maxDepth = 6
2021-11-08 11:10:12,517 [myid:] - INFO  [main:RequestPathMetricsCollector@112] - zookeeper.pathStats.initialDelay = 5
2021-11-08 11:10:12,517 [myid:] - INFO  [main:RequestPathMetricsCollector@113] - zookeeper.pathStats.delay = 5
2021-11-08 11:10:12,517 [myid:] - INFO  [main:RequestPathMetricsCollector@114] - zookeeper.pathStats.enabled = false
2021-11-08 11:10:12,518 [myid:] - INFO  [main:ZooKeeperServer@1536] - The max bytes for all large requests are set to 104857600
2021-11-08 11:10:12,519 [myid:] - INFO  [main:ZooKeeperServer@1550] - The large request threshold is set to -1
2021-11-08 11:10:12,519 [myid:] - INFO  [main:AuthenticationHelper@66] - zookeeper.enforce.auth.enabled = false
2021-11-08 11:10:12,519 [myid:] - INFO  [main:AuthenticationHelper@67] - zookeeper.enforce.auth.schemes = []
2021-11-08 11:10:12,519 [myid:] - INFO  [main:ZooKeeperServer@361] - Created server with tickTime 2000 minSessionTimeout 2000 maxSessionTimeout 2000 clientPortListenBacklog -1 datadir src\test\resources\data\test2414824585606454936.dir\version-2 snapdir src\test\resources\data\test2414824585606454936.dir\version-2
2021-11-08 11:10:12,535 [myid:] - INFO  [main:ZKTestCase@82] - FINISHED Optional[public void org.apache.zookeeper.server.quorum.LearnerTest.syncTest() throws java.lang.Exception]
